### PR TITLE
Added support for detecting the "networkidle" event

### PR DESCRIPTION
**Pull request by pdobrev**
Added support for detecting the "networkidle" event based on not having any network activity for 500ms (this is what Puppeteer uses).

Note that this implementation is not as complete as Puppeteer's because we don't track loads of scripts and image tags. We can enhance it with MutationObservers to handle that, but I opted to skip it for now, because:
- Generally the use of `networkidle` is discouraged, especially in Playwright's docs, so having a complete implementation is not that worth it. 
- Typically, what is being awaited are dynamic network calls to APIs and that we do cover.

Extending it will not be super hard, but I opted to leave that for later if the case ever comes.

### CODE REVIEW

File: Sources/sophiajs/src/sophia.ts [hunk 1]
```diff
@@ -611,13 +615,18 @@
...
 
-                // TODO: Add error handling here.
-                // req.addEventListener('error', function (event) {
-                //     var message = { payload: { url: (event?.currentTarget as any).responseURL } };
-                //     window.webkit.messageHandlers.handler.postMessage(message);
-                // });
+                req.addEventListener('error', function (_) {
+                    self.neworkIdleMonitor.onRequestEnd();
+                    // TODO: Add error handling here.
+                    // var message = { payload: { url: (event?.currentTarget as any).responseURL } };
```
Comment (ibratoev) [inline Sources/sophiajs/src/sophia.ts:621]: Remove comment out code.

File: TestServer/server.mjs [hunk 2]
```diff
@@ -52,6 +51,43 @@ app.use('/load_latency/assets/', (req, __, next) => {
...
+        next();
+        return;
+    }
+
+    setTimeout(() => {
+        next();
+    }, delay);
+});
+
+//
```
Comment (ibratoev) [inline TestServer/server.mjs:85]: cleanup

**Review by ibratoev — APPROVED** [review]
Nice tests. Added a few small cleanup comments.

File: TestServer/server.mjs (hunk 2 above)
Comment (pdobrev) [reply TestServer/server.mjs:85]: This was semi-intentional (it's just in the wrong place :)), but how would you go about separating different sections of the file without splitting it into separate files (e.g. here I wanted to have the group of load latency be visually separate from xhr latency group). I've seen folks do `////////////////////////////////////////////` but I don't like that. So I through just more whitespace between the sections.

I'm totally fine with not visually separating anything, it was more for aesthetics.

File: Sources/sophiajs/src/sophia.ts (hunk 1 above)
Comment (pdobrev) [reply Sources/sophiajs/src/sophia.ts:621]: will remove it, will leave the TODO, since it's actually not handled. need to make a note to cover that.

**Pull request merged**